src/Cassandra/TabletMap.cs

```diff
@@ -0,0 +1,170 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Cassandra
+{
+    public class TabletMapBackgroundUpdater
+    {
+        private readonly TabletMap _tabletMap;
+        private readonly BlockingCollection<TabletAddRequest> _queue = new BlockingCollection<TabletAddRequest>();
```

Comment:
Please generalize requests:
```suggestion
        private readonly BlockingCollection<TabletMapRequest> _queue = new BlockingCollection<TabletMapRequest>();
```